src/curl.c

```diff
@@ -62,6 +62,34 @@ static void swupd_curl_strerror(CURLcode curl_ret)
 	fprintf(stderr, "Curl error: (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
 }
 
+/*
+ * Cannot avoid a TOCTOU here with the current curl API.  Only using the curl
+ * API, a curl_easy_setopt does not detect if the client SSL certificate is
+ * present on the filesystem.  This only happens during curl_easy_perform.
+ * The emphasis is rather on how using an SSL client certificate is an opt-in
+ * function rather than an opt-out function.
+ */
+CURLcode swupd_curl_set_optional_client_cert(CURL* curl) {
+	CURLcode curl_ret = CURLE_OK;
+	char* client_cert_path;
+
+	client_cert_path = mk_full_filename(path_prefix, SSL_CLIENT_CERT);
+	if (access(client_cert_path, F_OK) == 0) {
```

Comment:
Earlier, I had SSL_CLIENT_CERT here instead of client_cert_path which caused the most recent Travis failures. Now, Travis test pass.